nextcord/enums.py

```diff
@@ -87,7 +88,7 @@ class EnumMeta(type):
         _enum_member_map_: ClassVar[Dict[str, Any]]
         _enum_value_map_: ClassVar[Dict[Any, Any]]
 
-    def __new__(cls, name, bases, attrs, *, comparable: bool = False):
+    def __new__(mcs, name, bases, attrs, *, comparable: bool = False):
```

Comment:
This is a metaclass so `mcs` might "fix an inconsistency".

Reply:
I think it can be left as `cls` as `mcs` would conflict with [PEP 8](https://peps.python.org/pep-0008/#function-and-method-arguments)

> Always use `cls` for the first argument to class methods.

Also, it's better not to change things unrelated to the main focus of the PR since it makes it harder to review.

It's better to create a separate branch and PR for each issue fixed.

A lot of refactoring and type stuff in one PR is fine, but it shouldn't be in the same PR as a new feature in my opinion.